app/controllers/application_controller.rb

```diff
@@ -6,5 +6,14 @@ class ApplicationController < ActionController::Base
   def set_current_user
     Current.user = User.find_by(id: session[:user_id]) if session[:user_id]
   end
-  
+
+  def auth_check
+    # allows only logged in user
+   @user = User.find_by(id: params[:id])  
+    if @user.nil? || Current.user.id != @user.id  
+      redirect_to root_path
+      flash[:notice] = "Invalid opertion performed. "
+    end
+  end
```

Comment:
What's the need of this now? Maybe you can shift this message to users controller if needed.